libvast/src/system/exporter.cpp

```diff
@@ -108,6 +107,13 @@ behavior exporter(stateful_actor<exporter_state>* self, expression expr,
       self->quit(msg.reason);
     }
   );
+  self->set_down_handler(
+    [=](const down_msg& msg) {
+      VAST_DEBUG(self, "received DOWN from", msg.source);
+      if (msg.reason == exit_reason::user_shutdown)
+        shutdown(self);
```

Comment:
> A follow up question (to be sure): CTRL + C'ing the SINK process (i.e. `vast export -c "..."`) prompts the NODE shutdown related the EXPORTERs with a `down_msg`?

What do you mean by *prompts the NODE shutdown related the EXPORTERs with a `down_msg`*?

> When should shutdown be called here. Previously, this wasn't called when a `down_msg` or `exit_msg` was received. For the continuous query, shutdown should be called when the SINK is shutdown by the user.

Yeah, previously `down_msg` wasn't handled at all, because EXPORTER wouldn't monitor anything. As you mention, the shutdown should take place for both `down_msg` and `exit_msg`: 

1. `exit_msg`: report statistics unless we got `exit_reason::kill`.
2. `down_msg`: for continuous queries, report statistics unless the `down_msg` came from a sink. Because all sinks are hidden behind a broadcasting `actor_pool`, we need to invert the condition: only report statistics if the `down_msg` came from ARCHIVE or INDEX. This means that if the historical option is set, EXPORTER should monitor ARCHIVE and INDEX as well.

> To prevent confusion with the other cases where shutdown is called, we could also extract the reporting functionality from `shutdown`  and call that here ...

I like this idea; let's call that function `report_statistics`.